native/data_dypes.md

```diff
@@ -0,0 +1,102 @@
+### Abstracting Data Types
+
+An initial effort to abstract the data types between JS engine(s) and addon(s).
+
+#### Definitions: (Alphabetical order)
+
+ - Addon or Native Addon : Node.JS modules developed using C,C++.
+ - Chakra : A Javascript engine developed by Microsoft
+ - ECMAScript : The scripting language standardized by Ecma International in the ECMA-262 specification and ISO/IEC 16262. (ref Wikipedia)
+ - IO.JS : [Bringing ES6 to the Node Community!](http://iojs.org)
+ - JS Engine : Javascript engine that powers Node.JS or it's forks (Chakra, SpiderMonkey, V8 ...)
+ - JXcore : [Node.JS on mobile and embedded devices](http://jxcore.io/)
+ - MDN : Mozilla Developers Network
+ - Node : [Node.JS](http://nodejs.org) and it's previously mentioned sub projects
+ - SpiderMonkey : A Javascript engine developed by Mozilla.
+ - V8 : A Javascript engine developed by Google.
+
+
+#### JS Types
+
+The latest ECMAScript standard defines seven data types: (ref. MDN and contributors [2015] [Online] Available from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures))
+
+    Six data types that are primitives:
+        Boolean
+        Null
+        Undefined
+        Number
+        String
+        Symbol (new in ECMAScript 6)
+    and 
+        Object
+
+Initial list of the corresponding native types are given below;
+
+```
+enum JSTypes {
+  Int32, // Number
+  Double, // Number
+  Boolean, // Boolean
+  String, // String
+  Object, // Object
+  Buffer, // Specialized Object
+  Undefined, // Undefined
+  Null, // Null
+  Error, // Specialized Object
+  Function // Callable Object
+};
+```
+
+`JSTypes type_;`
+
+Given list does not provide a definition for ObjectTemplate or FunctionTemplate V8 types. Assumption is that, these sub specializations can be provided via API methods. 
+
+`JS Engine` native type definition and / or behavior is engine and version specific. However, regardless from the implementation details, previously mentioned data types are available for all the mature JS engines.
+
+Following the **common** `JS engine` behavior, there are two important memory state a native type representation can be exist;
+
+ - Persistent / Rooted Variable
+ - Scoped / Temp Variable
+
+`Node` is not expected to manage the memory for JS Engine variable for any of the given options above. One of the [verified](https://github.com/jxcore/jxcore/blob/master/doc/native/Embedding_Basics.md) assumptions show that; a void ptr can point to a native JS Engine variable without dealing with its memory management.
+
+`void *data_;` -> JS Engine Variable
```

Comment:
From addon or core developer's point ? Cross engine / binary, JSValue needs to have a same byte order and size. Addon developer is not expected to be reaching any of the JSValue internals.  

Internally void* can be wrapped into something with name/type but don't know how much difference that would make. Just trying to avoid one extra wrapper. Suggesting on type/wrapper?